src/containers/changeAccountEmail/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { Helmet } from 'react-helmet';
+import { Alert, Button, Form, Input, Typography } from 'antd';
+import { ContentContainer } from '../../components';
+import ConfirmationMessage from '../../components/ConfirmationMessage';
+import { Link as RouterLink } from 'react-router-dom';
+import ProtectedApiClient from '../../api/protectedApiClient';
+import styled from 'styled-components';
+import { ChangeEmailRequest } from './ducks/types';
+
+const { Title, Link } = Typography;
+
+const UpdateButton = styled(Button)`
+  margin-top: 20px;
+`;
+
+const ChangeAccountEmail: React.FC = () => {
+  const [loading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [emailChanged, setEmailChanged] = useState<boolean>(false);
+
+  const onFinishChangeEmail = async (values: ChangeEmailRequest) => {
+    try {
+      setError(false);
+      setIsLoading(true);
+      await ProtectedApiClient.changeAccountEmail(values);
+      setEmailChanged(true);
+    } catch (err) {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
```

Comment:
```suggestion
  const [changeEmail, setChangeEmail] = useState<AsyncRequest<void, any>>(AsyncRequestNotStarted());

  const onFinishChangeEmail = async (values: ChangeEmailRequest) => {
    try {
      setChangeEmail(AsyncRequestLoading());
      await ProtectedApiClient.changeAccountEmail(values);
      setChangeEmail(AsyncRequestCompleted());
    } catch (err) {
      setChangeEmail(AsyncRequestFailed(err));
    } 
  };
```

Personally, I think this is a lot more readable.

Reply:
Yes!!! This should be in the docs! 😍

Reply:
Looks dope

Reply:
@jackblanc What should I pass into `AsyncRequestCompleted()` on line 24 of the suggested change?

Reply:
Just gonna pass it in undefined